huaweicloud/services/evs/data_source_huaweicloud_evs_volumes.go

```diff
@@ -0,0 +1,231 @@
+package evs
+
+import (
+	"context"
+	"strconv"
+
+	"github.com/chnsz/golangsdk/openstack/evs/v2/cloudvolumes"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/config"
+	"github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/helper/hashcode"
+	"github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/utils/fmtp"
+	"github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/utils/logp"
+)
+
+func DataSourceEvsVolumesV2() *schema.Resource {
+	return &schema.Resource{
+		ReadContext: dataSourceEvsVolumesV2Read,
+
+		Schema: map[string]*schema.Schema{
+			"region": {
+				Type:     schema.TypeString,
+				Optional: true,
+			},
+			"availability_zone": {
+				Type:     schema.TypeString,
+				Optional: true,
+			},
+			"dedicated_storage_id": {
+				Type:     schema.TypeString,
+				Optional: true,
+			},
+			"dedicated_storage_name": {
+				Type:     schema.TypeString,
+				Optional: true,
+			},
+			"enterprise_project_id": {
+				Type:     schema.TypeString,
+				Optional: true,
+			},
+			"shareable": {
+				Type:     schema.TypeBool,
+				Optional: true,
+			},
+			"server_id": {
+				Type:     schema.TypeString,
+				Optional: true,
+			},
+			"status": {
+				Type:     schema.TypeString,
+				Optional: true,
+			},
+			"volumes": {
+				Type:     schema.TypeList,
+				Computed: true,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"id": {
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+						"attachments": {
+							Type:     schema.TypeList,
+							Computed: true,
+							Elem: &schema.Resource{
+								Schema: map[string]*schema.Schema{
+									"id": {
+										Type:     schema.TypeString,
+										Computed: true,
+									},
+									"attached_at": {
+										Type:     schema.TypeString,
+										Computed: true,
+									},
+									"attached_mode": {
+										Type:     schema.TypeString,
+										Computed: true,
+									},
+									"device": {
```

Comment:
suggest name it as `device_name`, which is the same as in the console

Reply:
done